### PR TITLE
Remove useMedia3p hook

### LINKED ISSUE #3428: Remove useMedia3p and useMedia3pForProvider overloads
useMedia3p and useMedia3pForProvider are only used in Media3pPane, and so don't aid readability in any way.

### PR DESCRIPTION
**Pull request by cvolzke4**
## Summary

Remove useMedia3p and useMedia3pForProvider.
These are only used in Media3pPane, and so don't aid readability in any way.

## Testing Instructions

Open Unsplash media pane, scroll through a page of media, enter a search term, select a category.

Fixes #3428

### CODE REVIEW

File: assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js [hunk 1]
```diff
@@ -78,46 +75,42 @@ function Media3pPane(props) {
...
+      setSelectedProvider,
+      setSearchTerm,
+      unsplash,
+      selectedProviderState: media3p[selectedProvider ?? ProviderType.UNSPLASH],
     })
   );
 
+  const {
+    state: {
+      categories: { categories, selectedCategoryName },
```
Comment (diegovar) [inline assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js]: I wonder if this state should be called "category", with a "categories" and "selectedCategoryName" in it.

File: assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js [hunk 2]
```diff
@@ -78,46 +75,42 @@ function Media3pPane(props) {
...
+    unsplash,
+    selectedProviderState,
+  } = useMedia(
+    ({
+      media3p: {
+        state: { selectedProvider, searchTerm },
+        actions: { setSelectedProvider, setSearchTerm },
+        unsplash,
+      },
+      media3p,
```
Comment (diegovar) [inline assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js:91]: Does js not allow destructuring here, like 
media3p[selectedProvider ?? ProviderType.UNSPLASH]: {
  //needed state
}
?

File: assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js [hunk 3]
```diff
@@ -149,20 +142,20 @@ function Media3pPane(props) {
...
-          resources={media}
-          isMediaLoading={isMediaLoading}
-          isMediaLoaded={isMediaLoaded}
-          hasMore={hasMore}
+          resources={unsplash.state.media}
+          isMediaLoading={unsplash.state.isMediaLoading}
+          isMediaLoaded={unsplash.state.isMediaLoaded}
+          hasMore={unsplash.state.hasMore}
+          setNextPage={unsplash.actions.setNextPage}
           onInsert={insertMediaElement}
```
Comment (diegovar) [inline assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js:157]: should we use selectedProvider an selectedProviderState here? Then we don't need to extract unsplash above.

**Review by winterDroid — APPROVED** [review]
I like that change 👍

File: assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js (hunk 2 above)
Comment (cvolzke4) [reply assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js:91]: This isn't possible here, as 'selectedProvider' is not yet declared.

File: assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js (hunk 3 above)
Comment (cvolzke4) [reply assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js:157]: Coverr will be a separately rendered PaginatedMediaGallery.

**Comment by cvolzke4** [comment]
> I like that change 👍

I like this also, thanks for suggesting it.

File: assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js (hunk 1 above)
Comment (cvolzke4) [reply assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js]: It appears unusual to have categories.categories, but I still think the plural makes sense. The attribute still stores 'categories', in a plural context.

This is what the state structure looks like in context:
https://screenshot.googleplex.com/xZ7HOFgtKYE

Comment (diegovar) [reply assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js]: I understand the rationale, I still find it confusing and error prone though in a world without types. How about categoryState?

Comment (cvolzke4) [reply assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js]: I'll let Marc comment further, as he in working on categories.

Comment (winterDroid) [reply assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js]: I also didn't really like `categories.categories`. I thought about `categories.list` instead

**Pull request merged**